pkg/common/utils.go

```diff
@@ -17,3 +23,44 @@ func RemoveStringSliceItem(item string, slice *[]string) {
 		}
 	}
 }
+
+// ParseResponseForKeywords parses the response from detector verification calls for expected keywords in the response.
+func ParseResponseForKeywords(reader io.ReadCloser, keywords []string) (bool, error) {
```

Comment:
This won't perform as expected because reading from the body consumes the reader. That means that once you start reading it, you won't be able to read those parts again. Since we only need to check the response for one substring anyway, I recommend removing this function and just renaming `containsSubstring` to `ResponseContainsSubstring`